forge/csrc/passes/extract_unique_op_configuration.cpp

```diff
@@ -207,12 +206,79 @@ void export_unique_op_configuration_to_csv_file(
     fs.close();
 }
 
+void export_unique_op_configuration_to_xlsx_file(
+    const UniqueOpShapesAttrsType &unique_op_shapes_attrs, std::string graph_name, std::string stage)
+{
+    std::string export_unique_op_config_default_path = std::filesystem::current_path().string();
+    std::string export_unique_op_config_path =
+        env_as<std::string>("FORGE_EXPORT_UNIQUE_OP_CONFIG_DIR_PATH", export_unique_op_config_default_path);
+    export_unique_op_config_path = export_unique_op_config_path + "/OpConfigs/" + graph_name + "/";
+    if (not std::filesystem::exists(std::filesystem::path(export_unique_op_config_path)))
+    {
+        TT_ASSERT(
+            std::filesystem::create_directories(std::filesystem::path(export_unique_op_config_path)),
+            "Export Directory creation failed!");
+    }
+    std::string export_unique_op_config_file = export_unique_op_config_path + stage + ".xlsx";
+
+    py::module my_module = py::module::import("forge.utils");
+
+    py::str py_export_unique_op_config_file_path = export_unique_op_config_file.c_str();
+    py::str py_graph_name = graph_name.c_str();
+    py::str py_stage = stage.c_str();
+    py::dict py_unique_op_shapes_attrs;
+
+    for (const auto &[op_name, shapes_attrs] : unique_op_shapes_attrs)
```

Comment:
Can we try not to use auto when we don't need to. It makes things easier when we have actual type here

Reply:
Sure I will update it

Reply:
Replaced auto with actual type